components/embedder_support/BUILD.gn

```diff
@@ -13,7 +13,7 @@ source_set("unit_tests") {
   deps = [
     "//base/test:test_support",
     "//components/embedder_support",
-    "//components/embedder_support:browser_util",
+    "//components/embedder_support:user_agent",
```

Comment:
why do we have our own `embedder_support` component? That makes no sense in Brave.

Reply:
it doesn't make sense because `embedder_support` is for other projects that use chromium so whatever we subclass in here should just go in brave/browser

Reply:
For Desktop this is only here for the unit test because it's related to `chromium_src/components/embedder_support/user_agent_utils.cc` override.
As for Android cc: @samartnik

Reply:
On android we extend `org/chromium/components/embedder_support/view/ContentView`, but we don't create a separate component for it, we add our `BraveContentView.java` into upstream's component. There is also `BraveUrlConstants`, but it doesn't extend anything and definitely can be moved elsewhere.

Reply:
I see, in the meantime have we not made any progress on what the issue is? It's concerning to me that this is causing a crash otherwise

Reply:
`BraveUrlConstants` definitely does not belog here though and should be moved

Reply:
Not sure about progress, @SergeyZhukovsky worked on it, but he is on PTO at the moment, so he probably won't be able to update on it till Monday
I'll move `BraveUrlConstants`

Reply:
Moved it here https://github.com/brave/brave-core/commit/53ed0a9e3b5f3ac5f2171c9b9db5b99380849f31

Reply:
@bridiver the crash fix is still in progress, I still hope to get reliable steps for it. The temp fix I did it seems like removed the original crash in that place but it fails in a different right now with Android OS stack only unfortunately. There are some steps from a user https://github.com/brave/brave-browser/issues/41185#issuecomment-2720779843, I'm going to try them. And the user mentioned that they can replicate it in Chrome as well with an original stack that we removed https://github.com/brave/brave-browser/issues/41185#issuecomment-2720807967